doc/api.yml

```diff
@@ -0,0 +1,844 @@
+          schema:
+            $ref: '#/components/schemas/TooManyRequestsException'
+    '500':
+      description: Internal server error
+      content:
+        application/json:
+          schema:
+            $ref: '#/components/schemas/InternalServerException'
+
+  schemas:
+    ## enum type
+    Page:
+      type: integer
+      description: Start at `1`
+    PageSize:
+      type: integer
+      description: Cannot exceed `500`
+    Sex:
+      type: string
+      enum:
+        - M
+        - F
+        - OTHER
+    ReactionType:
+      type: string
+      enum:
+        - LIKE
+        - DISLIKE
+    UserStatus:
+      type: string
+      enum:
+        - ENABLED
+        - BANISHED
+    PostStatus:
+      type: string
+      enum:
+        - ARCHIVED
+        - ENABLED
+        - DRAFT
+
+        ## resource model
+    Category:
+      properties:
+        id:
+          type: string
+        label:
+          type: string
+    Post:
+      properties:
+        id:
+          type: string
+        picture_url:
+          type: string
+        description:
+          type: string
+        content:
+          type: string
+        title:
+          type: string
+        creation_datetime:
+          type: string
+          format: date-time
+        reaction_count:
+          type: integer
+        updated_at:
+          type: string
+          format: date-time
+        user_id:
+          type: string
+    Comment:
+      properties:
+        id:
+          type: string
+        post_id:
+          type: string
+        comment:
+          type: string
+        creation_datetime:
+          type: string
+          format: date-time
+        reply_to_comment_id:
+          type: string
+    Reaction:
+      properties:
+        id:
+          type: string
+        type:
+          $ref: '#/components/schemas/ReactionType'
+        creation_datetime:
+          type: string
+          format: date-time
+        user_id:
+          type: string
+        post_id:
+          type: string
+        comment_id:
+          type: string
+    User:
+      allOf:
+        - $ref: '#/components/schemas/UserInfo'
+      properties:
+        id:
+          type: string
+        photo_url:
+          type: string
+        bio:
+          type: string
+        profile_banner_url:
+          type: string
+        username:
+          type: string
+        about:
+          type: string
+        status:
+          $ref: '#/components/schemas/UserStatus'
+        sex:
+          $ref: '#/components/schemas/Sex'
+        entrance_datetime:
+          type: string
+          format: date-time
+    UserInfo:
+      properties:
+        last_name:
+          type: string
+        first_name:
+          type: string
+        birth_date:
+          type: string
+          format: date
+        email:
+          type: string
+    LoginInUser:
+      properties:
+        email:
+          type: string
+        password:
+          type: string
+        userId:
```

Comment:
Just email and password because Firebase will return the `userId`

Reply:
We need `userId` (Firebase Unique ID) to be able to associate a given firebase user to our user record
Firebase does return the userId but since we start auth initialization on the front end

we require email & password to strenghten our authentication mecanism